Fixed golint errors for pkg/kubelet/dockershim/libdocker

shimmy568 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

This PR fixed linting issues in `pkg/kubelet/dockershim/libdocker`

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Part of #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

/priority backlog
/area code-organization
/area kubelet

shimmy568 committed:
Fixed golint errors for pkg/kubelet/dockershim/libdocker

mattjmcnaughton commented on pkg/kubelet/dockershim/libdocker/fake_client.go:
```diff
@@ -832,13 +868,15 @@ func (f *FakeDockerClient) updateContainerStatus(id, status string) {
 	}
 }
 
+// ResizeExecTTY is a fake implemenattion of the call in the real client, the call is stored in the fake with the name "resize_exec"
```
Nit: `s/implemenattion/implementation`

mattjmcnaughton commented on pkg/kubelet/dockershim/libdocker/fake_client.go:
```diff
@@ -903,10 +943,12 @@ type FakeDockerPuller struct {
 	client Interface
 }
 
+// Pull pulls\ an image from the fake client that stored in the fake puller
```
Nit: `s/pulls\/pulls`

mattjmcnaughton commented on pkg/kubelet/dockershim/libdocker/client.go:
```diff
@@ -27,16 +27,21 @@ import (
+	// MinimumDockerAPIVersion is the lower docker version acceptable
 	// https://docs.docker.com/engine/reference/api/docker_remote_api/
-	// docker version should be at least 1.13.1
+	// docker version should be at least this high
```
Just double checking: Does the git history show that `MinimumDockerAPIVersion` used to be 1.13.1, but someone updated it to 1.26.0 and forgot to update the comment?

shimmy568 replied:
It was updated in this (f62cc81934634433eb8c7dbfc5bf755247a8efeb). It seems that 1.13.1 is the docker version while the 1.26.0 is the API version. I'll update the comment so this distinction is a little more clear to the reader.

mattjmcnaughton reviewed (commented):
Thank you for all your work here @shimmy568 :)

A couple small comments from me. Once they addressed, feel free to tag me again and I'll mark as lgtm.

shimmy568 committed:
Updated comments for issues brought up in PR

Fixed spelling in comments for PR and updated comments
for min docker API version to be more clear

mattjmcnaughton reviewed (approved):
/lgtm

Thank you for addressing my comments :) LGTM!

shimmy568 commented:
@mattjmcnaughton I don't think I changed anything that effects the code. Was the test failure a fluke or is there something else going on here?

mattjmcnaughton commented:
> @mattjmcnaughton I don't think I changed anything that effects the code. Was the test failure a fluke or is there something else going on here?

Looks like the `kubernetes-verify` failure is because of a spelling error - see https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/85112/pull-kubernetes-verify/1194964235109535744.

My guess is the `node-e2e-containerd` issue is flake. Triggering a retest which will give us more info :)
/test pull-kubernetes-node-e2e-containerd

tanjunchen commented:
/test pull-kubernetes-verify
/test pull-kubernetes-node-e2e-containerd

mattjmcnaughton reviewed (approved):
/test pull-kubernetes-verify

/lgtm

LGTM assuming the verify check passes (which I believe it should after you corrected the misspelling).

mattjmcnaughton reviewed (changes requested):
/lgtm cancel

Whoops, I misread and assumed the spelling errors were already corrected. Please ping me when they are fixed and I can mark lgtm.

shimmy568 committed:
Fixed spelling errors in comments

shimmy568 commented:
@mattjmcnaughton Fixed the spelling mistakes. Sorry for the wait

mattjmcnaughton reviewed (approved):
/lgtm

LGTM assuming the test suite passes.

tanjunchen commented:
/test pull-kubernetes-e2e-gce

ash2k commented:
@shimmy568 this needs to be rebased as there is a conflict now.
/ok-to-test
